src/mjolnir/shortcutbuilder.cc

```diff
@@ -527,6 +533,17 @@ uint32_t AddShortcutEdges(GraphReader& reader,
       // Compute the weighted edge density
       newedge.set_density(average_density / (static_cast<float>(length)));
 
+      // Update speed to the one that takes turn durations into account
+      assert(total_duration > 0);
```

Comment:
Would it make sense to be more direct here and either log an error or raise an exception? Since this is not in the hot-loop at runtime, we can take more care to find issues with the input data.

Reply:
yeah i suggested to use the assert. i dont think there is any way that we set a 0 speed on directed_edges which is why i think an assert is ok. basically it says this is likely to never happen. remember that we only make shortcuts on level 0 and 1 edges which are all motorvehicle access only and im not sure we should ever get to the point where we are storing those types of edges with a speed of 0

Reply:
oh i thought this was one of the other asserts above. i guess in this case it would only happen if the edges length were 0, which again is one of those things that we have a bit of code in there that says length must be at least 1